apps/dashboard-service/src/dashboard/warmup/dashboard-cache-warmup.service.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable } from "@nestjs/common";
+import { Cron, CronExpression } from "@nestjs/schedule";
+import { CacheService } from "../dto/cache.service";
+import { TotalSectionHeaderService } from "../dto/total-section-header.service";
+import { TreeRestorationGoalService } from "../dto/tree-restoration-goal.service";
+import { DashboardQueryDto } from "../dto/dashboard-query.dto";
+import { InjectRedis } from "@nestjs-modules/ioredis";
+import Redis from "ioredis";
+import { TMLogger } from "@terramatch-microservices/common/util/tm-logger";
+
+@Injectable()
+export class DashboardCacheWarmupService {
+  private readonly logger = new TMLogger(DashboardCacheWarmupService.name);
+
+  constructor(
+    private readonly cacheService: CacheService,
+    private readonly totalSectionHeaderService: TotalSectionHeaderService,
+    private readonly treeRestorationGoalService: TreeRestorationGoalService,
+    @InjectRedis() private readonly redis: Redis
+  ) {}
+
+  @Cron(CronExpression.EVERY_WEEK)
+  async warmup() {
+    this.logger.log("Starting dashboard cache warmup...");
+
+    try {
+      this.logger.log("Cleaning up dashboard keys...");
+      await this.cleanupDashboardKeys();
+      this.logger.log("Dashboard keys cleaned");
+
+      const landscapesList = ["gcb", "grv", "ikr"] as const;
+      const organisationTypesList = ["non-profit-organization", "for-profit-organization"] as const;
+      const cohortsList = ["terrafund", "terrafund-landscapes"] as const;
+
+      const generateCombinations = <T>(items: readonly T[]): T[][] => {
+        const combos: T[][] = [];
+        const count = items.length;
+        combos.push([]);
+        for (let i = 1; i < 1 << count; i++) {
+          const combo: T[] = [];
+          for (let j = 0; j < count; j++) {
+            if ((i & (1 << j)) !== 0) combo.push(items[j]);
+          }
+          combos.push(combo);
+        }
+        return combos;
+      };
```

Comment:
There doesn't seem to be any reason to define this method within the surrounding method. Please move to the top of the file above the class definition.